### PR TITLE
Drop unneeded capture groups

### PR DESCRIPTION
**Pull request by fgsch**
Follow up of #1111.

This covers all the remaining rules.  Minor style changes while here.

For the curious, for this PR I wrote a small program that prints the number of capture groups using `PCRE_INFO_CAPTURECOUNT` and run it through all the patterns to have a list of rules that needed reviewing.

### CODE REVIEW

**Comment by fgsch** [comment]
For the curious, the last one is:

```
"(?i:([\s'\"`\(\)]*?)([\d\w]++)([\s'\"`\(\)]*?)(?:<(?:=(?:([\s'\"`\(\)]*?)(?!\2)([\d\w]+)|>([\s'\"`\(\)]*?)(?:\2))|>?([\s'\"`\(\)]*?)(?!\2)([\d\w]+))|(?:not\s+(?:regexp|like)|is\s+not|>=?|!=|\^)([\s'\"`\(\)]*?)(?!\2)([\d\w]+)|(?:(?:sounds\s+)?like|r(?:egexp|like)|=)([\s'\"`\(\)]*?)(?:\2)))"
```

Rule 942130.

**Comment by emphazer** [comment]
@fgsch did you test the regexp assemble result with a diff?

**Comment by fgsch** [comment]
@emphazer What do you mean?

**Comment by emphazer** [comment]
i ment the ./regexp-assemble.pl result of regexp-942130.data and regexp-942180.data.

**Comment by fgsch** [comment]
I got that part :) What do you mean with `test the result with a diff`?

FWIW, the result of the regexp assemble is what's in the rules. I did not change the pattern manually for the generated ones but pasted the result.

**Comment by emphazer** [comment]
i wasn't sure if you have to backtick some parts.
your work looks good so far. :)

**Comment by csanders-git** [comment]
This looks good so far but there are a lot of changes.

**Comment by fgsch** [comment]
@csanders-git I agree. Maybe people can comment below each change?

**Comment by csanders-git** [comment]
Good idea!!

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 1]
```diff
@@ -454,7 +454,7 @@ SecRule REQUEST_HEADERS:Content-Type "@rx ^(?:application\/x-www-form-urlencoded
     ver:'OWASP_CRS/3.0.0',\
     severity:'WARNING',\
     chain"
-    SecRule REQUEST_BODY|XML:/* "@rx \%((?!$|\W)|[0-9a-fA-F]{2}|u[0-9a-fA-F]{4})" \
+    SecRule REQUEST_BODY|XML:/* "@rx \%(?:(?!$|\W)|[0-9a-fA-F]{2}|u[0-9a-fA-F]{4})" \
```
Comment (csanders-git) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf:445]: clearly equivalent... G2G

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 2]
```diff
@@ -1194,7 +1194,7 @@ SecRule REQUEST_BASENAME "@endsWith .pdf" \
         setvar:'tx.%{rule.id}-OWASP_CRS/PROTOCOL_VIOLATION/INVALID_HREQ-%{matched_var_name}=%{matched_var}'"
 
 
-SecRule ARGS "@rx \%((?!$|\W)|[0-9a-fA-F]{2}|u[0-9a-fA-F]{4})" \
+SecRule ARGS "@rx \%(?:(?!$|\W)|[0-9a-fA-F]{2}|u[0-9a-fA-F]{4})" \
```
Comment (csanders-git) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf:1233]: clearly equivalent... G2G

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf [hunk 3]
```diff
@@ -305,7 +305,7 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
     setvar:'tx.anomaly_score=+%{tx.critical_anomaly_score}',\
     setvar:'tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/SQLI-%{matched_var_name}=%{tx.0}'"
 
-SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?i:(?:(union(.*?)select(.*?)from)))" \
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?i:)union.*?select.*?from" \
```
Comment (csanders-git) [inline rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:299]: Seems to be equivalent, good to go.

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf [hunk 4]
```diff
@@ -513,7 +513,7 @@ SecRule TX:PARANOIA_LEVEL "@lt 2" "id:942014,phase:2,pass,nolog,skipAfter:END-RE
 # Identifies common initial SQLi probing requests where attackers insert/append
 # quote characters to the existing normal payload to see how the app/db responds.
 #
-SecRule ARGS_NAMES|ARGS|XML:/* "@rx (^\s*[\"'`;]+|[\"'`]+\s*$)" \
+SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?:^\s*[\"'`;]+|[\"'`]+\s*$)" \
```
Comment (csanders-git) [inline rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:501]: Seems to be equivalent, good to go.

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf [hunk 5]
```diff
@@ -587,9 +587,9 @@ SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?i:(?:(?:^|\W)in[+\s]*\([\s\d\"]+[^()]*\)|\
 #   ./regexp-assemble.pl regexp-942130.data
 # Note that after assemble an outer bracket with an ignore case flag is added
 # to the Regexp::Assemble output:
-#   (?i:ASSEMBLE_OUTPUT)
+#   (?i)ASSEMBLE_OUTPUT
```
Comment (csanders-git) [inline rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:573]: this is a comment...?

Comment (fgsch) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:573]: Yes, this is to keep the comment in sync with the change below.

Comment (franbuehler) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:573]: Comment was in sync with the rule below. No change needed here.

Comment (fgsch) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:573]: @franbuehler No, I changed the regexp. Before:

```(?i:([...```

Now:

```(?i)[...```

So I updated the comment to be in sync with my change.

**Comment by dune73** [comment]
This is a welcome PR. However, it has not been merged in months and the review is still not complete. We have therefore taken a closer look during the community chat on Slack last night.

The PR aims for 3.1, but given the change is not directly relevant to users and given it is a quite a big change coming in after the RC, we will not take it into the 3.1 release. 

This means, the PR will need to have moved to 3.2/dev.  Maybe @franbuehler wants to do the review.

**Comment by lifeforms** [comment]
- @fgsch will rebase the PR on `v3.2/dev`
- @franbuehler will review the regexps

**Comment by fgsch** [comment]
Rebased against v3.2/dev.

**Comment by fgsch** [comment]
Superseded by #1252 and upcoming PRs.
Leaving it open until all the changes in this PR are covered.

**Comment by fgsch** [comment]
For the record here is the list of all the rules with capture groups:

```
 Rule 901420 has 1 groups
 Rule 901440 has 1 groups
 Rule 910140 has 1 groups
 Rule 912150 has 1 groups
 Rule 920230 has 1 groups
 Rule 920240 has 1 groups
 Rule 920440 has 1 groups
 Rule 920480 has 1 groups
 Rule 921180 has 1 groups
 Rule 931130 has 1 groups
 Rule 932140 has 1 groups
 Rule 942110 has 1 groups
 Rule 942180 has 1 groups
 Rule 942250 has 1 groups
 Rule 942420 has 1 groups
 Rule 942421 has 1 groups
 Rule 942430 has 1 groups
 Rule 942431 has 1 groups
 Rule 942432 has 1 groups
 Rule 943110 has 1 groups
 Rule 951240 has 1 groups
 Rule 901430 has 2 groups
 Rule 920190 has 2 groups
 Rule 942440 has 2 groups
 Rule 942270 has 3 groups
 Rule 942130 has 11 groups
 Rule 941190 has 17 groups
 Rule 941250 has 25 groups
 Rule 941330 has 60 groups
 Rule 941220 has 136 groups
 Rule 941210 has 168 groups
```

**Comment by fgsch** [comment]
The remaining rules except for the last 2 are covered in #1257.
Closing.

**Comment by dune73** [comment]
How about the remaining two?

**Comment by fgsch** [comment]
Once the outstanding PR is merged I will open a new PR for those.

**Pull request closed without merging**